libexec/bats-core/bats-exec-test

```diff
@@ -115,8 +115,16 @@ bats_exit_trap() {
     if [[ -n "$line" ]]; then
       printf '# %s\n' "$line"
     fi
+    
+    if [[ ( -z "${BATS_TRACE_LEVEL}" || "${BATS_TRACE_LEVEL}" -gt 0 ) && -n "${BATS_TRACE_OUTPUT_COMBINED[*]:-}" ]]; then
+      printf "# %s\n" "${BATS_TRACE_OUTPUT_COMBINED[@]}" >&3
+    fi
+
     status=1
   else
+    if [[ "${BATS_TRACE_LEVEL}" -gt 0 ]]; then
+      printf "   %s\n" "${BATS_TRACE_OUTPUT_COMBINED[@]}" >&3
```

Comment:
Why omit the leading `#`? This will break tap output.

Also, why print that before the `ok` but the other one after the `not ok`?
I think that should be unified and probably moved behind the if. However, I'd vote for only printing this output in failure cases.